src/sentry/migrations/0437_backfill_notificationsetting.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 2.2.28 on 2023-05-08 15:23
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+ACTOR_TEAM_TYPE = 0
+ACTOR_USER_TYPE = 1
+
+
+def backfill_notificationsetting(apps, schema_editor):
+    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
+    Actor = apps.get_model("sentry", "Actor")
+
+    # Only modify records that don't have a user or team set.
+    queryset = NotificationSetting.objects.filter(user_id__isnull=True, team_id__isnull=True)
```

Comment:
This will select ~15k rows.

Reply:
And this is a `OR` filter?

Reply:
No, this is an AND filter. We need to update rows that are missing both a team_id and user_id. If a notificationsetting record has either column filled it is ok.

Reply:
gooooot it yeh.